### PR TITLE
[fix](glue)support access glue and s3 iceberg with credential list

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

support access glue and s3 iceberg with credential list

support  iceberg hadoop catalog on s3

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/s3_util.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ bool S3ClientFactory::is_s3_conf_valid(const std::map<std::string, std::string>&
 }
 
 bool S3ClientFactory::is_s3_conf_valid(const S3Conf& s3_conf) {
-    return !s3_conf.ak.empty() && !s3_conf.sk.empty() && !s3_conf.endpoint.empty();
+    return !s3_conf.endpoint.empty();
```
Comment (morningman) [inline be/src/util/s3_util.cpp:138]: Why only endpoint is required?

Comment (wsjz) [reply be/src/util/s3_util.cpp:138]: Because we must set it to aws config:
aws_config.endpointOverride = s3_conf.endpoint;

**Comment by wsjz** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
compile fail

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by wsjz** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**